src/Types/ScriptLookups.purs

```diff
@@ -1058,6 +1060,25 @@ processConstraint mpsMap osMap = do
         -- constraint already.
         _cpsToTxBody <<< _outputs %= Array.(:) txOut
         _valueSpentBalancesOutputs <>= provideValue amount
+    MustPayToScriptInlineDatum vlh dat plutusValue -> do
+      -- TODO Check if vlh corresponds to v1 or v2?
```

Comment:
I guess we can't check for this particular constraint. I imagine the error from the evaluate or submit endpoints would point users to the problem if they tried to use inline datums with `PlutusV1` scripts

Reply:
Indeed I tested it at some point, it failed with a pretty good message. Maybe I'll write a test to make sure it does fail rather than submit

Reply:
Actually it fails when trying to *spend*, meaning the value is locked forever. Not great as you kinda realise too late

Reply:
Oh that's not ideal

Reply:
https://github.com/cardano-foundation/CIPs/pull/310/files#diff-4ffe445f8e72a6cee701a5a989aef62750d96c6296058bb612cad8ff305b49a5R24 touches on this exact issue actually